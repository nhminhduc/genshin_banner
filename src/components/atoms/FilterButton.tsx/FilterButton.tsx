import { useEffect, useState } from "react";

type FilterButtonProp = {
  checked?: boolean,
  className?: string,
  id: string,
  label: string,
  name: string,
  onButtonClick(element: string, isChecked: boolean): void;
}

const FilterButton = ({
  checked = false,
  className,
  id,
  label,
  name,
  onButtonClick,
}: FilterButtonProp) => {
  const [isChecked, setIsChecked] = useState(checked);

  const handleOnChange = () => {
    setIsChecked((prev) => !prev);
  };

  useEffect(() => {
    onButtonClick(id, isChecked)
  }, [isChecked]);

  return (
    <div className={className}>
      <input
        id={id}
        name={name}
        onChange={handleOnChange}
        type="checkbox"
        checked={isChecked}
      />
      <label htmlFor={id}>{label}
      </label>
    </div>
  );
};

export default FilterButton;
