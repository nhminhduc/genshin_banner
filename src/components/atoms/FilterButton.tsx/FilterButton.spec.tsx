import { render } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";

import FilterButton from "./FilterButton";

describe("FilterButton test", () => {
  test("FilterButton load", () => {
    const mockFunction = vi.fn();
    const { container } = render(
      <FilterButton
        id="FilterButton-test"
        name="FilterButton test"
        onButtonClick={mockFunction}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
