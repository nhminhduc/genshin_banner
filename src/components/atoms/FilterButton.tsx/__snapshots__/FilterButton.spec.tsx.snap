// Vitest Snapshot v1

exports[`FilterButton test > FilterButton load 1`] = `
<div>
  <button
    class="flex justify-around items-center"
    id="FilterButton-test"
    name="FilterButton test"
    type="button"
  >
     
  </button>
</div>
`;
